python/llm/src/ipex_llm/transformers/npu_models/convert.py

```diff
@@ -455,7 +455,7 @@ def optimize_llm_single_process(
 
 def prepare_input_ids(
         self, input_ids, past_key_values=None, attention_mask=None, inputs_embeds=None, **kwargs):
-    if past_key_values is not None:  # kvcache
+    if isinstance(past_key_values, bool) and past_key_values:  # kvcache
```

Comment:
`if past_key_values and isinstance(past_key_values, bool)`?

Reply:
Updated.